addons/event_sale/models/event_registration.py

```diff
@@ -97,7 +97,8 @@ def write(self, vals):
     def _synchronize_so_line_values(self, so_line):
         if so_line:
             return {
-                'partner_id': False if self.env.user._is_public() else so_line.order_id.partner_id.id,
+                # Avoid registering public users but respect the portal workflows
+                'partner_id': False if self.env.user._is_public() and self.env.user.partner_id == so_line.order_id.partner_id else so_line.order_id.partner_id.id,
```

Comment:
Hi @chienandalu, i'm experiencing the same issue in Odoo saas~16.4+e (Enterprise Edition) in production.

While your solution works, I have a suggestion to make it simpler:

In the case of a public user registering for a ticket on website, the sale order would have customer set to public user.

Instead of checking if transaction user is public, we could check if the sale order user is public. 

```suggestion
                'partner_id': False if so_line.order_id.partner_id.is_public else so_line.order_id.partner_id.id,
```

Reply:
Where do you get that `is_public` flag from?

Reply:
I was going to propose the same thing, indeed it is not defined in 15. Note that this "is_public" is False is there is no user on the partner, so I think it is not completely correct.